src/factories.js

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import components from './index';
+
+let factories = {};
+
+Object.keys(components).forEach(function (name) {
+  if (components[name] instanceof React.Component.constructor) {
```

Comment:
Great, that gives me a nice way forward. I'll work on this and bundle everything together. Thanks!